packages/components/src/color-palette/index.native.js

```diff
@@ -110,7 +121,7 @@ function ColorPalette( {
 
 	function isSelectedCustom() {
 		const isWithinColors =
-			activeColor && mergedColors && mergedColors.includes( activeColor );
+			activeColor && allColors && allColors.includes( activeColor );
```

Comment:
Nitpick: we could simplify it by removing one of the conditions.
```suggestion
			activeColor && allColors?.includes( activeColor );
```

Reply:
Nice! https://github.com/WordPress/gutenberg/pull/57605/commits/dd7b91a4df40180b6bb9d420bb94081679bef46e